### PR TITLE
Handsontable should work properly in <iframe>

### PR DESCRIPTION
**Pull request by swistach**
### Context
Currently, we use global `window` and `document` to do every DOM operation what could make for bugs.

### How has this been tested?
Use the following example https://jsfiddle.net/cagrLk4z/

### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)

### Related issue(s):
1. #5686

### CODE REVIEW

File: src/helpers/feature.js [hunk 1]
```diff
@@ -38,6 +39,17 @@ export function requestAnimationFrame(callback) {
   return _requestAnimationFrame.call(window, callback);
 }
 
+export function isClassListSupported() {
+  return !!document.documentElement.classList;
+}
+
+export function isTextContentSupported() {
+  return document.createTextNode('test').textContent;
```
Comment (budnix) [inline src/helpers/feature.js]:
```suggestion
  return !!document.createTextNode('test').textContent;
```
Missing to Boolean conversion.

Comment (swistach) [reply src/helpers/feature.js]: You're right! Thanks @budnix :+1:

File: src/plugins/copyPaste/focusableElement.js [hunk 2]
```diff
@@ -170,10 +172,13 @@ function destroyElement(wrapper) {
     refCounter = 0;
 
     // Detach secondary element from the DOM.
+    const secondaryElement = secondaryElements.get(wrapper.rootDocument);
+
     if (secondaryElement && secondaryElement.parentNode) {
       secondaryElement.parentNode.removeChild(secondaryElement);
-      secondaryElement = null;
+      secondaryElements.set(wrapper.rootDocument, null);
```
Comment (wszymanski) [inline src/plugins/copyPaste/focusableElement.js]: I think that [Map.prototype.delete()](https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/Map/delete) / [WeakMap.prototype.delete()
](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap/delete) should be used here.

File: src/plugins/copyPaste/focusableElement.js [hunk 3]
```diff
@@ -118,34 +119,35 @@ function forwardEventsToLocalHooks(eventManager, element, subject) {
   eventManager.addEventListener(element, 'paste', runLocalHooks('paste', subject));
 }
 
-let secondaryElement;
+const secondaryElements = new Map();
```
Comment (wszymanski) [inline src/plugins/copyPaste/focusableElement.js]: Probably the `WeakMap` will be a better choice here.

File: src/renderers/autocompleteRenderer.js [hunk 4]
```diff
@@ -26,14 +17,18 @@ clonableARROW.appendChild(document.createTextNode(String.fromCharCode(9660)));
  * @param {Object} cellProperties Cell properites (shared by cell renderer and editor)
  */
 function autocompleteRenderer(instance, TD, row, col, prop, value, cellProperties, ...args) {
+  const { rootDocument } = instance;
   const rendererType = cellProperties.allowHtml ? 'html' : 'text';
-  const ARROW = clonableARROW.cloneNode(true); // this is faster than createElement
+  const ARROW = rootDocument.createElement('DIV'); // this is faster than createElement
```
Comment (wszymanski) [inline src/renderers/autocompleteRenderer.js]: Should it be changed? If so, please remove the comment.

File: src/core.js [hunk 5]
```diff
@@ -90,14 +90,29 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
...
 
-  keyStateStartObserving();
+  this.rootElement = rootElement;
+  /**
+   * The nearest document over container.
+   *
+   * @private
+   * @type {Document}
+   */
+  this.rootDocument = rootElement.ownerDocument;
```
Comment (wszymanski) [inline src/core.js]: Maybe add here comment that it's a reference to the same element as in the Walkontable. Any `TODO` comment informing that it should be changed.

File: src/helpers/unicode.js [hunk 6]
```diff
@@ -117,7 +117,7 @@ export function isMetaKey(keyCode) {
 export function isCtrlKey(keyCode) {
   const keys = [];
 
-  if (window.navigator.platform.includes('Mac')) {
+  if (navigator.platform.includes('Mac')) {
```
[the remark this answers is not in the document]
Comment (swistach) [reply src/helpers/unicode.js]: `window` is not necessary 🙂

File: src/3rdparty/walkontable/src/overlay/_base.js [hunk 7]
```diff
@@ -229,10 +235,14 @@ class Overlay {
         preventOverflow === 'vertical' && this.type === Overlay.CLONE_LEFT) {
       this.mainTableScrollableElement = rootWindow;
 
+    } else if (rootWindow.getComputedStyle(wtTable.wtRootElement.parentNode).getPropertyValue('overflow') === 'hidden') {
+      this.mainTableScrollableElement = wtTable.holder;
     } else {
       this.mainTableScrollableElement = getScrollableElement(wtTable.TABLE);
     }
 
+    // console.log(this.mainTableScrollableElement);
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/overlay/_base.js]: Please remove this comment.

File: src/3rdparty/walkontable/src/overlay/left.js [hunk 8]
```diff
@@ -146,8 +146,7 @@ class LeftOverlay extends Overlay {
    */
   adjustRootElementSize() {
     const { wtTable, rootDocument, rootWindow } = this.wot;
-    const masterHolder = wtTable.holder;
-    const scrollbarHeight = masterHolder.clientHeight === masterHolder.offsetHeight ? 0 : getScrollbarWidth(rootDocument);
+    const scrollbarHeight = getScrollbarWidth(rootDocument);
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/overlay/left.js]: Is name of variable proper?

File: src/helpers/dom/element.js [hunk 9]
```diff
@@ -643,7 +649,7 @@ export function getScrollableElement(element) {
       computedOverflowY = computedStyle.getPropertyValue('overflow-y');
       computedOverflowX = computedStyle.getPropertyValue('overflow-x');
 
-      if (computedOverflow === 'scroll' || computedOverflowX === 'scroll' || computedOverflowY === 'scroll') {
+      if (props.indexOf(computedOverflow) !== -1 || props.indexOf(computedOverflowX) !== -1 || props.indexOf(computedOverflowY) !== -1) {
```
Comment (wszymanski) [inline src/helpers/dom/element.js]: Maybe [Array.prototype.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) will be yet more clear?

File: src/3rdparty/walkontable/src/overlay/left.js (hunk 8 above)
Comment (swistach) [reply src/3rdparty/walkontable/src/overlay/left.js]: In this context it's ok. I think we can leave it at it is.

**Pull request merged**